File: frontend/webshop/src/app/models/product.ts
import { AverageRatingAndCount } from "./average-rating-and-count";
import { Brand } from "./brand";
import { Category } from "./category";
import { ProductNote } from "./product-note";
import { Review } from "./review";
import { SizePrice } from "./size-price";
import { Type } from "./type";

export class Product {
  productId: number;
  name: string;
  category: Category;
  type: Type;
  brand: Brand;
  description: string;
  imageUrl: string;
  sizePrices: SizePrice[];
  productNotes: ProductNote[];
  reviews?: Review[];
  averageRatingAndCount?: AverageRatingAndCount;
  amount?: number;
}

File: backend/webshop/product/src/main/java/com/fer/hr/product/model/Product.java
package com.fer.hr.product.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.engine.jdbc.Size;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;

    private String name;
    private String description;
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "type_id", nullable = false)
    private Type type;

    @ManyToOne
    @JoinColumn(name = "brand_id", nullable = false)
    private Brand brand;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<SizePrice> sizePrices;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductNote> productNotes;

    // Getters and Setters
}

